demo/json_demo.py

```diff
@@ -0,0 +1,52 @@
+"""This file provides preliminary methods of accessing and storing JSON data."""
+import json
+import os
+
+
+def main():
+    """Pull data from a provided json, and write it back to the json."""
+    user_data_dict = get_dict_from_json_file("demofile")
+    print("Dictionary:\n")
+    for username in user_data_dict:
+        print(username + ":", user_data_dict[username], "\n")
+    json_format_string = write_dict_to_json_file(user_data_dict, "demofile")
+    print("Json-formatted string:", "\n", json_format_string, "\n")
+    new_user = {"testuser": {"commits": [], "issues": [], "pull_requests": []}}
+    json_format_string = add_user_to_json(
+        user_data_dict, new_user, "testuser", "demofile"
+    )
+    print("Json-formatted string:", "\n", json_format_string, "\n")
+
+
+def get_dict_from_json_file(json_file_name):
+    """Populate and return a dictionary of all the data in a specified json file."""
+    with open(os.path.join("./data/", json_file_name + ".json"), "r") as json_file:
+        # In the open() function, "r" specifies read-only access
+        json_string = json_file.read()
+        user_data_dict = json.loads(json_string)
+        # json.loads() converts a json-formatted string into a python dictionary
+    return user_data_dict
+
+
+def write_dict_to_json_file(user_data_dict, json_file_name):
+    """Overwrite specified json file with data from a given dictionary."""
+    with open(os.path.join("./data/", json_file_name + ".json"), "w") as json_file:
+        # In the open() function, "w" specifies write access
+        json_string = json.dumps(user_data_dict, indent=4)
+        # json.dumps() converts a dictionary into a json-formatted string
+
+        # Specifying an indent does not alter the data itself, it only increases
+        # readability in the json file.
+
+        json_file.write(json_string)
+    return json_string
+
+
+def add_user_to_json(user_data_dict, to_add, key_to_add, json_file_name):
+    """Append data to the dictionary and write it to json"""
+    user_data_dict[key_to_add] = to_add[key_to_add]
+    return write_dict_to_json_file(user_data_dict, json_file_name)
```

Comment:
This is a good point... it may simply be better to only add the user to the dictionary, and then manually call the function to write it to the file. My thinking here was that it would keep the database updated, but during the runtime of the Cogitate tool it likely would only be accessed a few times, and the rest of the modifications would be done to a dictionary. I believe that I will change this function to simply add to the dictionary.

Reply:
Sounds good!